src/data.rs

```diff
@@ -135,21 +135,15 @@ pub fn get_confidence_from_db(db: Arc<DB>, block_number: u32) -> Result<Option<u
 		.cf_handle(crate::consts::CONFIDENCE_FACTOR_CF)
 		.context("Couldn't get column handle from db")?;
 
-	let res = db
-		.get_cf(&cf_handle, block_number.to_be_bytes())
-		.context("Couldn't get confidence in db")?;
-	let res = res.map(|data| {
-		data.try_into()
-			.map_err(|_| anyhow!("Conversion failed"))
-			.context("Unable to convert confindence (wrong number of bytes)")
-			.map(u32::from_be_bytes)
-	});
-
-	match res {
-		Some(Ok(r)) => Ok(Some(r)),
-		None => Ok(None),
-		Some(Err(e)) => Err(e),
-	}
+	db.get_cf(&cf_handle, block_number.to_be_bytes())
```

Comment:
This change makes the code less explicit. Any optimisation stemming from it does not outweigh the time needed to read the logic.

Reply:
What part of it? Removing 2 `let res = ...` or adding `.transpose()` instead of match block?